easybuild/easyblocks/l/lammps.py

```diff
@@ -224,46 +347,87 @@ def configure_step(self, **kwargs):
             self.cfg.update('configopts', '-DGPU_ARCH=%s' % get_cuda_gpu_arch(cuda_cc))
 
+        # Make sure that all libraries end up in the same folder (python libs seem to default to lib, everything else
+        # to lib64)
+        self.cfg.update('configopts', '-DCMAKE_INSTALL_LIBDIR=lib')
+
         # avoid that pip (ab)uses $HOME/.cache/pip
         # cfr. https://pip.pypa.io/en/stable/reference/pip_install/#caching
         env.setvar('XDG_CACHE_HOME', tempfile.gettempdir())
         self.log.info("Using %s as pip cache directory", os.environ['XDG_CACHE_HOME'])
 
+        # Make sure it uses the Python we want
+        python_dir = get_software_root('Python')
+        if python_dir:
+            python_short_version = '.'.join(get_software_version('Python').split('.')[:2])
+            python_lib = glob.glob("%s/lib*/libpython%s.so" % (python_dir, python_short_version))[0]
+            # Whether you need one or the other of the options below depends on the version of CMake and LAMMPS
+            # Rather than figure this out, use both (and one will be ignored)
+            self.cfg.update('configopts', '-DPython_EXECUTABLE=%s/bin/python' % python_dir)
+            self.cfg.update('configopts', '-DPYTHON_EXECUTABLE=%s/bin/python' % python_dir)
+            # Older LAMMPS need more hints to get things right as they use deprecated CMake packages
+            self.cfg.update('configopts', '-DPYTHON_LIBRARY=%s' % python_lib)
+            self.cfg.update('configopts', '-DPYTHON_INCLUDE_DIR=%s/include' % python_dir)
+
         return super(EB_LAMMPS, self).configure_step()
 
+    def install_step(self):
+        """Install LAMMPS and examples/potentials."""
+        super(EB_LAMMPS, self).install_step()
+        # Copy over the examples so we can repeat the sanity check
+        # (some symlinks may be broken)
+        examples_dir = os.path.join(self.start_dir, 'examples')
+        copy_dir(examples_dir, os.path.join(self.installdir, 'examples'), symlinks=True)
+        potentials_dir = os.path.join(self.start_dir, 'potentials')
+        copy_dir(potentials_dir, os.path.join(self.installdir, 'potentials'))
+
     def sanity_check_step(self, *args, **kwargs):
         """Run custom sanity checks for LAMMPS files, dirs and commands."""
+
+        # Output files need to go somewhere (and has to work for --module-only as well)
+        execution_dir = tempfile.mkdtemp()
+
         check_files = [
             'atm', 'balance', 'colloid', 'crack', 'dipole', 'friction',
-            'hugoniostat', 'indent', 'melt', 'message', 'min', 'msst',
+            'hugoniostat', 'indent', 'melt', 'min', 'msst',
             'nemd', 'obstacle', 'pour', 'voronoi',
         ]
 
         custom_commands = [
             # LAMMPS test - you need to call specific test file on path
             """python -c 'from lammps import lammps; l=lammps(); l.file("%s")'""" %
             # The path is joined by "build_dir" (start_dir)/examples/filename/in.filename
```

Comment:
The `build_dir` in this line confuses me. Certainly needs updating for `start_dir` becoming `installdir` below.

Reply:
Tweaked